src/analysis/liveness-transfer-function.h

```diff
@@ -7,35 +7,45 @@
 namespace wasm::analysis {
 
 class LivenessTransferFunction : public Visitor<LivenessTransferFunction> {
-  FinitePowersetLattice::Element currState;
+  FinitePowersetLattice::Element* currState;
 
 public:
-  LivenessTransferFunction(FinitePowersetLattice& lattice)
-    : currState(lattice.getBottom()) {}
+  LivenessTransferFunction() : currState(nullptr) {}
 
   // Transfer function implementation. Modifies the state for a particular
   // expression type. In our current limited implementation, we just update
   // state on gets and sets of local indices.
-  void visitLocalSet(LocalSet* curr) { currState.set(curr->index, false); }
-  void visitLocalGet(LocalGet* curr) { currState.set(curr->index, true); }
+  void visitLocalSet(LocalSet* curr) {
+    if (currState) {
```

Comment:
This can probably be an assertion instead of a check, since there should always be a `currState` when we're visiting expressions.